### PR TITLE
Parallelization 

### PR DESCRIPTION
**Pull request by gw2princeps**
Adds a second optimizer core, which is implemented in rust, fast, multithreaded and has heuristics to allow calculating billions of extras combinations in reasonable time. 

The rust implementation must be enabled manually in the settings. 

Details https://sopuli.xyz/post/713722

### CODE REVIEW

File: generateRustMappings.mts
Comment (gw2princeps) [inline generateRustMappings.mts:1]: this is not used at the moment, but gonna leave it for now

**Comment by gw2princeps** [comment]
Re CI failure: I had to change the package.json type to module (for some reason). I think it was related to the wasm tooling otherwise not working correctly.

**Comment by marcustyphoon** [comment]
I don't remember the exact status on this, but if we could make all of the tests pass, keep the Rust mode default-off, and hard-disable heuristic mode until we figure out the bug in it, I'd be happy to merge this in.

Would be great to include tests that put the same input into both core implementations and ensure that they output builds in the same order, but that would require implementing result deduplication in the Rust mode. As a step one, maybe a test that checks a given test input against an expected test output?

**Comment by gw2princeps** [comment]
Done so far, but I cant reproduce the eslint issue that is still thrown. I added `wasm_module` to the excluded section for eslint, but for some reason it still tries to scan it in the CI

**Comment by marcustyphoon** [comment]
I fixed the eslint issue.

Right now, the file renames to .cjs are not complete, making `yarn extractLocale` broken. There is also a merge conflict due to one of the cjs renames. There are two ways to fix these:
 - Continue migrating to `type: module` (#578) and make sure the upstream changes to testModifiers.js don't get lost in the merge resoution.
 - Don't migrate to `type: module` by making generateRustMappings commonJS for now (#579), which will make no merge conflict happen.

My guess is that the second one is easier for now? We can always consider migrating to `type: module` at a later point, but not doing so now will make this diff cleaner.

**Comment by gw2princeps** [comment]
If I remember correctly, I spent a fair bit of time avoiding `type: module`, but I couldn't get the rust tooling to work without esm. If you can find a quick solution, I don't mind opting for 2.

**Comment by marcustyphoon** [comment]
does #579 have the problems you encountered or are they fixed?

**Comment by gw2princeps** [comment]
The rust mappings file is not really used any more. Asides from that, it appears to work. 

My issue was probably related to something else! Sometimes hard to identify the exact root cause of a problem in the JS ecosystem

**Review by marcustyphoon — APPROVED** [review]

**Pull request merged**